sched/signal/sig_deliver.c

```diff
@@ -182,9 +185,10 @@ void nxsig_deliver(FAR struct tcb_s *stcb)
        * in the current sigprocmask that were already set by newsigprocmask.
        */
 
-      altsigprocmask    = stcb->sigprocmask ^ newsigprocmask;
-      stcb->sigprocmask = (stcb->sigprocmask & altsigprocmask) |
-                          (savesigprocmask & ~altsigprocmask);
+      nxsig_xorset(&tmpset1, &stcb->sigprocmask, &newsigprocmask);
+      sigandset(&tmpset2, &stcb->sigprocmask, &tmpset1);
+      nxsig_nandset(&tmpset3, &savesigprocmask, &tmpset1);
+      sigorset(&stcb->sigprocmask, &tmpset2, &tmpset3);
```

Comment:
```suggestion
      sigorset(&stcb->sigprocmask, &tmpset2, &tmpset3);
```
Just to make sure this is not forgotten